api/src/pcapi/scripts/transform_virtual_venues/main.py

```diff
@@ -0,0 +1,294 @@
+            enseigne="Enseigne",
+            raison_sociale="Raison sociale",
+        )
+
+
+def get_backend() -> EntrepriseWithHeadQuartersBackend | EntrepriseWithHeadQuartersTestingBackend:
+    if settings.IS_PROD or settings.IS_STAGING:
+        return EntrepriseWithHeadQuartersBackend()
+    return EntrepriseWithHeadQuartersTestingBackend()
+
+
+def get_head_quarter_info(siren: str) -> HeadQuarterInfo:
+    return get_backend().get_head_quarter(siren)
+
+
+def get_offerer_with_one_virtual_venue_query() -> BaseQuery:
+    sub_query = (
+        sqla.select(sqla_func.count(Venue.id).label("venue_count"), Venue.managingOffererId)
+        .select_from(Venue)
+        .group_by(Venue.managingOffererId)
+        .subquery("sub_venues")
+    )
+
+    return (
+        Offerer.query.filter(sub_query.c.venue_count == 1, Venue.isVirtual == True)
+        .join(sub_query, sub_query.c.managingOffererId == Offerer.id)
+        .join(Venue, Venue.managingOffererId == Offerer.id)
+    )
+
+
+def get_address(offerer: Offerer) -> Address:
+    """Search address based on offerer's geographical data.
+
+    If nothing is found, use the BAN API and create the missing address.
+    """
+    addresses = search_addresses(street=offerer.street or "", city=offerer.city, postal_code=offerer.postalCode)
+
+    if addresses:
+        return addresses[0]
+
+    ban_address = api_adresse.get_address(
+        address=offerer.street,
+        postcode=offerer.postalCode,
+        city=offerer.city,
+        strict=True,
+    )
+    location_data = LocationData(
+        street=ban_address.street,
+        city=ban_address.city,
+        postal_code=ban_address.postcode,
+        insee_code=ban_address.citycode,
+        latitude=ban_address.latitude,
+        longitude=ban_address.longitude,
+        ban_id=ban_address.id,
+    )
+
+    return get_or_create_address(location_data=location_data, is_manual_edition=False)
+
+
+def transform_virtual_venue_to_physical_venue(venue: Venue, offerer: Offerer, dry_run: bool) -> None:
+    # Fetching administrative data on the Entreprise API
+    if not offerer.siren:
+        raise MissingSirenError()
+
+    head_quarter_info = get_head_quarter_info(offerer.siren[:9])
+
+    if not (head_quarter_info.active and head_quarter_info.diffusible):
+        raise SiretNotActiveOrNotDiffusible()
+
+    # Fetching address data
+    address = get_address(offerer)
+
+    venue.street = address.street  # type: ignore[method-assign]
+    venue.departementCode = address.departmentCode
+    venue.postalCode = address.postalCode
+    venue.city = address.city
+    venue.latitude = address.latitude
+    venue.longitude = address.longitude
+    venue.offererAddress = get_or_create_offerer_address(offerer_id=offerer.id, address_id=address.id)
+    venue.isVirtual = False
+    venue.siret = head_quarter_info.siret
+    venue.venueTypeCode = VenueTypeCode.ADMINISTRATIVE
+    venue.publicName = head_quarter_info.enseigne or offerer.name
```

Comment:
Le nom d'usage n'est rempli que s'il est différent du nom "administratif", je ne vois pas de raison de le remplir ou modifier s'il n'y a pas d'enseigne :
```suggestion
    if head_quarter_info.enseigne:
        venue.publicName = head_quarter_info.enseigne
```